hackathon_site/dashboard/frontend/src/api/types.ts

```diff
@@ -42,3 +42,45 @@ export interface Category {
     max_per_team: number;
     unique_hardware_count: number;
 }
+
+/** Event API */
+export interface Profile {
+    id: number;
+    id_provided: boolean;
+    attended: boolean;
+    acknowledge_rules: boolean;
+    e_signature: string;
+    team: number;
+}
+
+export interface User {
+    id: number;
+    first_name: string;
+    last_name: string;
+    email: string;
+    profile: Profile;
+}
+
+type UserWithoutProfile = Omit<User, "profile">;
+type ProfileWithoutTeamNumber = Omit<Profile, "team">;
+
+export interface Team {
+    id: number;
+    team_code: string;
+    created_at: string;
+    updated_at: string;
+    profiles: ({ profile: ProfileWithoutTeamNumber } & { user: UserWithoutProfile })[];
```

Comment:
I think this will create
`profiles: [ { profile: ProfileWithoutTeam, user: UserWithoutProfile}]`
but we want
`profiles: [ { ... <properties ofProfileWithoutTeam>, user: UserWithoutProfile}]`
so maybe try:
```suggestion
    profiles: (ProfileWithoutTeamNumber & { user: UserWithoutProfile })[];
```